core/arch/arm/mm/pgt_cache.c

```diff
@@ -340,6 +341,139 @@ void pgt_flush_ctx(struct tee_ta_ctx *ctx)
 	mutex_unlock(&pgt_mu);
 }
 
+static void flush_pgt_entry(struct pgt *p)
+{
+	tee_pager_pgt_save_and_release_entries(p);
+	assert(!p->num_used_entries);
+	p->ctx = NULL;
+	p->vabase = 0;
+}
+
+static bool pgt_entry_matches(struct pgt *p, void *ctx, vaddr_t begin,
+			      vaddr_t last)
+{
+	if (!p)
+		return false;
+	if (p->ctx != ctx)
+		return false;
+	if (last > begin)
```

Comment:
Hmm, this looks wrong. Shouldn't it be `if (last <= begin)` (taking into account the case where `begin == last`)?